### PR TITLE
Add missing <cstddef> includes for size_t

### PR DESCRIPTION
**Pull request by guitargeek**
#### PR description:

To make cmssw compile with gcc 11.

#### PR validation:

Building CMSSW with gcc 11 gives errors without this PR, but succeeds with this PR.

### CODE REVIEW

**Comment by qliphy** [comment]
please test

File: DataFormats/SiPixelDigi/interface/SiPixelDigisSoA.h [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #ifndef DataFormats_SiPixelDigi_interface_SiPixelDigisSoA_h
 #define DataFormats_SiPixelDigi_interface_SiPixelDigisSoA_h
 
+#include <cstddef>
```
Comment (VinInn) [inline DataFormats/SiPixelDigi/interface/SiPixelDigisSoA.h:4]: I pretend that the correct action is to change size_t in uint32_t

**Comment by civanch** [comment]
+1

**Comment by civanch** [comment]
@VinInn , this is not the only place where size_t is used. I do not know arguments pro and contr but if size_t should be substituted by uint32_t, then this should be a campaign to substitute everywhere.

**Comment by qliphy** [comment]
+1

**Pull request merged**